boa/src/exec/statement_list.rs

```diff
@@ -0,0 +1,42 @@
+//! Statement list execution.
+
+use super::{Context, Executable, InterpreterState};
+use crate::{syntax::ast::node::StatementList, BoaProfiler, Result, Value};
+
+impl Executable for StatementList {
+    fn run(&self, interpreter: &mut Context) -> Result<Value> {
+        let _timer = BoaProfiler::global().start_event("StatementList", "exec");
+
+        // https://tc39.es/ecma262/#sec-block-runtime-semantics-evaluation
+        // The return value is uninitialized, which means it defaults to Value::Undefined
+        let mut obj = Value::default();
+        interpreter
+            .executor()
+            .set_current_state(InterpreterState::Executing);
+        for (i, item) in self.statements().iter().enumerate() {
+            let val = item.run(interpreter)?;
+            match interpreter.executor().get_current_state() {
+                InterpreterState::Return => {
+                    // Early return.
+                    obj = val;
+                    break;
+                }
+                InterpreterState::Break(_label) => {
+                    // Early break.
+                    break;
+                }
+                InterpreterState::Continue(_label) => {
+                    break;
+                }
+                InterpreterState::Executing => {
+                    // Continue execution
+                }
+            }
+            if i + 1 == self.statements().len() {
+                obj = val;
+            }
+        }
+
+        Ok(obj)
+    }
+}
```

Comment:
I think you accidentaly re-introduced this file while merging @georgeroman, you've moved it here: https://github.com/georgeroman/boa/blob/afe9daf3fa07cdd6346f29907df56b6fd7bd7c78/boa/src/syntax/ast/node/statement_list/mod.rs#L59-L95
I think you should delete this file, right?

Reply:
GitHub says my comment is outdated, but I can see the file here: https://github.com/georgeroman/boa/blob/move_exec_to_ast/boa/src/exec/statement_list.rs

Reply:
Yes, sorry, I forgot to delete it.